### DISCUSSION TITLE
What if i don't want to use modules_statuses.json file?

### DISCUSSION THREAD

**Discussion by subscriptionflow-developer** [nWidart/laravel-modules#1387] [Q&A] [answered]
We have a feature to dynamically create custom modules with CRUD features in our app and we don't want to write modules_statuses.json file every time a new module is created. Is there any way to:

1. Exclude this file and make all modules enabled by default ?
2. Use database table instead of file to maintain module statuses?

Writing modules_statuses.json file is not a good idea for us because of mult-tenant architecture.

**Comment by sprklinginfo** [comment]
Sorry, this is not an answer but rather a question :P
I am working on a multi-tenant setup. I hope I can dynamically disable/enable modules for each domain when the application is run on different domains. Any insights to achieve it? Thank you!

**Comment by nghyane** [comment, marked as answer]
> We have a feature to dynamically create custom modules with CRUD features in our app and we don't want to write modules_statuses.json file every time a new module is created. Is there any way to:
> 
> 1. Exclude this file and make all modules enabled by default ?
> 2. Use database table instead of file to maintain module statuses?
> 
> Writing modules_statuses.json file is not a good idea for us because of mult-tenant architecture.


create custom base on  Nwidart\Modules\Contracts\FileActivator and config modules.activators

```
    'activators' => [
        'file' => [
            'class' => FileActivator::class,
            'statuses-file' => base_path('modules_statuses.json'),
            'cache-key' => 'activator.installed',
            'cache-lifetime' => 604800,
        ],
    ],
```
